dotnet/src/Agents/Abstractions/Definition/AgentDefinition.cs

```diff
@@ -0,0 +1,183 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Collections.Generic;
+using Microsoft.SemanticKernel.Agents.Definition;
+
+namespace Microsoft.SemanticKernel.Agents;
+
+/// <summary>
+/// Defines an agent.
+/// </summary>
+public sealed class AgentDefinition
+{
+    /// <summary>
+    /// Gets or sets the version of the schema being used.
+    /// </summary>
+    public string? Version
+    {
+        get => this._version;
+        set
+        {
+            Verify.NotNull(value);
+            this._version = value;
+        }
+    }
+
+    /// <summary>
+    /// Gets or sets the id of the deployed agent.
+    /// </summary>
+    public string? Id
+    {
+        get => this._id;
+        set
+        {
+            Verify.NotNull(value);
+            this._id = value;
+        }
+    }
+
+    /// <summary>
+    /// Gets or sets the type of the  agent.
+    /// </summary>
+    public string? Type
+    {
+        get => this._type;
+        set
+        {
+            Verify.NotNull(value);
+            this._type = value;
+        }
+    }
+
+    /// <summary>
+    /// Gets or sets the name of the  agent.
+    /// </summary>
+    public string? Name
+    {
+        get => this._name;
+        set
+        {
+            Verify.NotNull(value);
+            this._name = value;
+        }
+    }
+
+    /// <summary>
+    /// Gets or sets the description of the agent.
+    /// </summary>
+    public string? Description
+    {
+        get => this._description;
+        set
+        {
+            Verify.NotNull(value);
+            this._description = value;
+        }
+    }
+
+    /// <summary>
+    /// Gets or sets the system instructions for the agent to use.
```

Comment:
IMHO, with latest changes from system to developer roles, better remove `system`.

```suggestion
    /// Gets or sets the instructions for the agent to use.
```